pkg/controller/database/database_controller.go

```diff
@@ -76,18 +98,120 @@ type ReconcileDatabase struct {
 // +kubebuilder:rbac:groups=databases.schemahero.io,resources=databases,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=databases.schemahero.io,resources=databases/status,verbs=get;update;patch
 func (r *ReconcileDatabase) Reconcile(request reconcile.Request) (reconcile.Result, error) {
-	instance, err := r.getInstance(request)
+	databaseInstance, err := r.getInstance(request)
 	if err != nil {
 		return reconcile.Result{}, err
 	}
 
+	ctx := context.Background()
+
 	// A "database" object is realized in the cluster as a deployment object,
 	// in the namespace specified in the custom resource,
 
 	logger.Debug("reconciling database",
-		zap.String("name", instance.Name))
+		zap.String("name", databaseInstance.Name))
 
-	// TODO add watches back in if we need them.  Today they aren't used for anything
+	statefulsetName := fmt.Sprintf("%s-controller", databaseInstance.Name)
+	schemaHeroManagerImage := fmt.Sprintf("%s:%s", r.managerImage, r.managerTag)
+
+	vaultAnnotations, err := databaseInstance.GetVaultAnnotations()
+
+	if err != nil {
+		logger.Error(err)
+		return reconcile.Result{}, err
+	}
+	if vaultAnnotations == nil {
+		vaultAnnotations = map[string]string{}
+	}
+
+	if err := r.reconcileRBAC(ctx, databaseInstance); err != nil {
+		logger.Error(err)
+		return reconcile.Result{}, nil
+	}
+
+	// TODO detect k8s version and use appsv1 or appsv1beta
```

Comment:
This needs to detect the k8s version / api-resources to make the correct StatefulSet resource